vendor/github.com/moby/buildkit/solver/result/result.go

```diff
@@ -1,13 +1,12 @@
 package result
 
 import (
-	"reflect"
 	"sync"
 
 	"github.com/pkg/errors"
 )
 
-type Result[T any] struct {
+type Result[T comparable] struct {
```

Comment:
```
# github.com/moby/buildkit/frontend/gateway/client
vendor/github.com/moby/buildkit/frontend/gateway/client/client.go:18:29: Reference does not implement comparable
note: module requires Go 1.20
```

If we want to keep this change, we should update `vendor.mod` to `go 1.20`, right?

Reply:
Ah, good catch! I should have tested with an older Go compiler.

Reply:
I guess that's https://github.com/moby/moby/pull/46069 👀

Reply:
Ah, yes, I decided not to _force_ go1.21 in that PR (as most of the code definitely would still work on older versions); perhaps we still should? At least setting the `go.mod` to go1.20 could be reasonable I guess. (with go1.19 being EOL); at least for the master / v25.0-dev branch.

Reply:
I'll make it a separate commit, as we don't need it for the 24.0 (and other branches)

I guess in that case we might as well make it a separate PR; want to open one?

Reply:
https://github.com/moby/moby/pull/46562